app/models/storage_service.rb

```diff
@@ -0,0 +1,22 @@
+class StorageService < ApplicationRecord
+  include NewWithTypeStiMixin
+  include ProviderObjectMixin
+  include AsyncDeleteMixin
+  include AvailabilityMixin
```

Comment:
Probably shouldn't start out using Availability on a new model since that is supposed to be replaced by SupportsFeature